lib/pure/collections/critbits.nim

```diff
@@ -8,10 +8,34 @@
 #
 
 ## This module implements a `crit bit tree`:idx: which is an efficient
-## container for a sorted set of strings, or for a sorted mapping of strings. Based on the excellent paper
-## by Adam Langley.
+## container for a sorted set of strings, or for a sorted mapping of strings. Based on the
+## [excellent paper by Adam Langley](https://www.imperialviolet.org/binary/critbit.pdf).
 ## (A crit bit tree is a form of `radix tree`:idx: or `patricia trie`:idx:.)
 
+runnableExamples:
```

Comment:
I know it's pre-existing but it's pretty bad for a top-level runnableExamples that's supposed to explain the main points. 

Any improvements welcome; eg using `toCritBitTree` etc; nothing should be redundant here (tests are there for that)